src/MongoDB/WriteConcern.c

```diff
@@ -145,7 +144,7 @@ static PHP_METHOD(WriteConcern, __construct)
 
 	switch (ZEND_NUM_ARGS()) {
 		case 3:
-			if (Z_TYPE_P(journal) != IS_NULL) {
+			if (journal && Z_TYPE_P(journal) != IS_NULL) {
```

Comment:
Why was this change necessary? Did `zend_parse_parameters` previously _always_ initialize `zval *journal` or what this a long-standing bug?

Reply:
Without the additional check I was presented with a compiler warning about `journal` potentially being uninitialised. Looking through the sources, we can see that [`zend_parse_arg_zval_deref`](https://heap.space/xref/PHP-8.1/Zend/zend_API.h?r=76fcd70c#2256) will always assign the `arg` pointer to `*dest` if `check_null` is false (as is the case for `journal`). I'm not entirely sure what `arg` would be for optional arguments that weren't provided, or why this wasn't an issue before, especially considering that the call stack always ends up in the above `zend_parse_arg_zval_deref` function. I lack the internals knowledge to figure that one out, hence the simple `NULL` check.